llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -274,14 +261,15 @@ Expected<std::string> assembleSpirv(const std::string &inFilename) {
   if (binSize < 0)
     return createResultError(Result::ErrorInvalidShader, Twine("Failed to assemble SPIR-V: \n") + log);
 
-  fwrite(spvBin.data(), 1, binSize, outFile);
+  unsigned *bin = new unsigned[binSize / sizeof(unsigned)]();
+  std::copy(spvBin.data(), spvBin.data() + binSize / sizeof(unsigned), bin);
```

Comment:
nit: use `llvm::copy`

Could we return a vector instead of `BinaryData`, so that the memory management becomes more obvious?

Reply:
Ah, I can now see that `cleanupCompileInfo` should do it as long as the (de-)allocation functions match. Could you add comments explaining that the caller is responsible for calling `delete`?

Reply:
Yes - that's correct. I've added a comment to make this clearer.

Reply:
I've updated to use llvm::copy - but it seems a little circular since llvm::copy just invokes std::copy after converting the range into a start and end iterator. (Using llvm::copy requires the use of make_range to construct a range from the start and end!)

Reply:
Sory, I missed your reply. I agree that the current version is a bit circular. I thought of something like:
```c++
const unsigned binSizeWords = binSize = / sizeof(unsigned);
spvBin.resize(binSizeWords);
unsigned *bin = new unsigned[binSizeWords]();
llvm::copy(spirvBin, bin);
```

The current version is also fine, feel free to leave this as-is.